src/picongpu/include/particles/ionization/byCollision/ThomasFermi/AlgorithmThomasFermi.hpp

```diff
@@ -41,118 +41,178 @@ namespace particles
 
-                /* requires mass density in g/cm^3 */
-                constexpr float_64 nAvogadro = SI::N_AVOGADRO;
-                constexpr float_64 convM3ToCM3 = 1.e6;
+            /* requires mass density in g/cm^3 */
+            constexpr float_64 nAvogadro = SI::N_AVOGADRO;
+            constexpr float_64 convM3ToCM3 = 1.e6;
 
-                float_64 const convToMassDensity = densityUnit * massNumber / nAvogadro / convM3ToCM3;
-                float_64 const massDensity = density * convToMassDensity;
+            float_64 const convToMassDensity = densityUnit * massNumber / nAvogadro / convM3ToCM3;
+            float_64 const massDensity = density * convToMassDensity;
 
-                constexpr float_64 invAtomicTimesMassNumber = float_64(1.) / (protonNumber * massNumber);
-                float_64 const R = massDensity * invAtomicTimesMassNumber;
+            constexpr float_64 invAtomicTimesMassNumber = float_64(1.) / (protonNumber * massNumber);
+            float_64 const R = massDensity * invAtomicTimesMassNumber;
 
-                float_64 const Q_1 = A * math::pow(R,B);
+            float_64 const Q_1 = A * math::pow(R,B);
 
-                float_64 const Q = math::pow(math::pow(R,C) + math::pow(Q_1,C), float_64(1.)/C);
+            float_64 const Q = math::pow(math::pow(R,C) + math::pow(Q_1, C), float_64(1.) / C);
 
-                float_64 const x = thomasFermi::TFAlpha * math::pow(Q,TFBeta_temp);
+            float_64 const x = thomasFermi::TFAlpha * math::pow(Q, TFBeta_temp);
 
-                /* Thomas-Fermi average ionization state */
-                float_64 const ZStar = protonNumber * x / (float_64(1.) + x + math::sqrt(float_64(1.) + float_64(2.)*x));
+            /* Thomas-Fermi average ionization state */
+            float_X const ZStar = static_cast< float_X >(
+                protonNumber * x / (
+                    float_64(1.) + x +
+                    math::sqrt( float_64(1.) + float_64(2.) * x )
+                )
+            );
 
-                /* integral part of the charge state */
-                float_64 intZStar;
-                /* fractional part of the charge state */
-                float_X fracZStar = static_cast<float_X>(math::modf(ZStar,&intZStar));
+            return ZStar;
+        }
+        /** Functor implementation
+         *
+         * Calling this functor gives a prediction for an integer number of new
+         * free macro electrons to create. This prediction is based on the
+         * average charge state in the Thomas-Fermi model.
+         * The functor calculates the integer number of bound electrons from
+         * this state by a Monte-Carlo step.
+         *
+         * \tparam ParticleType type of particle to be ionized
+         *
+         * \param ZStar average charge state in the Thomas-Fermi model
+         * \param parentIon particle instance to be ionized
+         * \param randNr random number
```

Comment:
return value description is missing, please do not forget the the the output range